numba/tests/test_pycc.py

```diff
@@ -233,6 +233,34 @@ def check_compile_for_cpu(self, cpu_name):
             self.assertPreciseEqual(res, 123 * 321)
             self.assertEqual(lib.multi.__module__, 'pycc_test_simple')
 
+    def test_cc_export_nogil(self):
+        f_sig = "void(int64[:], intp[:])"
+
+        cc_nogil = CC('pycc_test_simple_nogil')
+
+        @cc_nogil.export("f_nogil", f_sig, nogil=True)
+        def f_nogil(n, x):
+            s = 0.0
+            for _ in range(n):
+                s += x
+            return s
+
+        self.assertTrue(cc_nogil._release_gil)
+
+    def test_cc_export_gil(self):
+        f_sig = "void(int64[:], intp[:])"
+
+        cc_gil = CC('pycc_test_simple_nogil')
+
+        @cc_gil.export("f_nogil", f_sig)
+        def f_gil(n, x):
+            s = 0.0
+            for _ in range(n):
+                s += x
+            return s
+
+        self.assertFalse(cc_gil._release_gil)
+
```

Comment:
@stuartarchibald After discussing with @guilhermeleobas I implemented the approach to test the correctness of GIL flag according the user provided value of `nogil` parameter in `cc.export`. Please let me know if its okay.